frontend/src/metabase/dashboard/containers/DashboardApp.jsx

```diff
@@ -132,6 +132,10 @@ const DashboardApp = props => {
 
   useUnmount(props.reset);
 
+  // most browsers don't use a custom message anymore, just putting here to retain compatibility
+  // https://developer.mozilla.org/en-US/docs/Web/API/Window/beforeunload_event#compatibility_notes
+  useBeforeUnload(isEditing && isDirty, "You have unsaved changes.");
+
```

Comment:
Please wrap the string into `t` from `ttag` to make sure it can be translated:

```js
import { t } from "ttag";

onBeforeOnload(isEditing && isDirty, t`You have unsaved changes.`);
```

Reply:
Done 😄